dissect/target/tools/info.py

```diff
@@ -90,49 +90,51 @@ def main():
 
 
 def get_target_info(target: Target) -> dict[str, Union[str, list[str]]]:
-    return {
+    info = {
+        "disks": get_disks_info(target),
+        "volumes": get_volumes_info(target),
+        "children": get_children_info(target),
         "hostname": target.hostname,
-        "domain": target.domain,
+        "domain": None,
         "ips": target.ips,
         "os_family": target.os,
         "os_version": target.version,
         "architecture": target.architecture,
-        "language": target.language,
-        "timezone": target.timezone,
-        "install_date": target.install_date,
-        "last_activity": target.activity,
-        "disks": get_disks_info(target),
-        "volumes": get_volumes_info(target),
-        "children": get_children_info(target),
+        "language": None,
+        "timezone": None,
+        "install_date": None,
+        "activity": None,
     }
 
+    optional = ["domain", "language", "timezone", "install_date", "activity"]
+
+    for func in optional:
+        if target.has_function(func):
+            info[func] = getattr(target, func)
+
+    return info
+
 
 def print_target_info(target: Target) -> None:
     print(target)
-    print("\nDisks")
-    for d in target.disks:
-        print(f"- {str(d)}")
-    print("\nVolumes")
-    for v in target.volumes:
-        print(f"- {str(v)}")
-
-    children = list(target.list_children())
-    if children:
-        print("\nChildren")
-        for child in children:
-            print(f"- <Child type='{child.type}' path='{child.path}'>")
-
-    print()
-    print(f"Hostname      : {target.hostname}")
-    print(f"Domain        : {target.domain}")
-    print(f"IPs           : {', '.join(target.ips or ['None'])}")
-    print(f"OS family     : {target.os} ({target._os_plugin.__name__})")
-    print(f"OS version    : {target.version}")
-    print(f"Architecture  : {target.architecture}")
-    print(f"Language(s)   : {', '.join(target.language)}")
-    print(f"Timezone      : {target.timezone}")
-    print(f"Install date  : {target.install_date}")
-    print(f"Last activity : {target.activity}")
+
+    for name, value in get_target_info(target).items():
+        if name in ["disks", "volumes", "children"]:
+            if not any(value):
+                continue
+            print(f"\n{name.capitalize()}")
+            for i in value:
+                values = " ".join([f'{k}="{v}"' for k, v in i.items()])
+                print(f"- <{name.capitalize()} {values}>")
+            continue
+
+        if isinstance(value, list):
+            value = ", ".join(value)
+
+        if name == "hostname":
+            print()
+
+        print(f"{name.capitalize().replace('_', ' ')}" + (14 - len(name)) * " " + f" : {value}")
```

Comment:
Your suggestion would break backwards compatibility with the json and flow record output. We do use those output formats extensively in an automated fashion.

Reply:
Ah, check, I missed that part.